src/main/java/com/celements/model/access/object/DefaultXObjectHandler.java

```diff
@@ -0,0 +1,247 @@
+package com.celements.model.access.object;
+
+import static com.google.common.base.Preconditions.*;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.component.annotation.InstantiationStrategy;
+import org.xwiki.component.annotation.Requirement;
+import org.xwiki.component.descriptor.ComponentInstantiationStrategy;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.WikiReference;
+
+import com.celements.model.access.IModelAccessFacade;
+import com.celements.model.access.exception.ClassDocumentLoadException;
+import com.celements.model.classes.fields.ClassField;
+import com.celements.model.context.ModelContext;
+import com.celements.model.util.AdjustWikiFunction;
+import com.celements.model.util.References;
+import com.google.common.base.Function;
+import com.google.common.base.Optional;
+import com.google.common.base.Predicate;
+import com.google.common.base.Predicates;
+import com.google.common.collect.FluentIterable;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+
+@NotThreadSafe
+@Component
+@InstantiationStrategy(ComponentInstantiationStrategy.PER_LOOKUP)
+public class DefaultXObjectHandler implements XObjectHandler {
```

Comment:
split class in a XWikiXObjectHandler (do not deprecate but add a hint that better type safty can be achieved by using ClassFieldXObjectHandler) with Strings and Objects for key and values and a ClassFieldXObjectHandler only using ClassFields<T> and values T too in the FilterMap(s)